### PR TITLE
Note storybook tests

### PR DESCRIPTION
**Pull request by razianmm**
**Before tagging the team for a review, I have done the following:**

- [x] run `yarn all` locally: ensures that all tests pass, formatting is done, types pass, and builds pass
- [x] reviewed my code changes
- [x] updated the Deploy Preview link at the top of this PR (or remove it if not applicable)
- [x] updated the Figma referense link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] done an accessibility check on my work (tested with Chrome's `axe Dev Tools`, Mac's VoiceOver, etc.)
- [x] tagged `@avaya-dux/dux-design` if any visual changes have occurred
- [x] tagged `@avaya-dux/dux-devs`

ADD PR SUMMARY: Added Storybook tests to Note Component

**EXAMPLE IMAGE(S):**
Screenshot 2024-10-22 at 6 39 04 PM

## Summary by CodeRabbit

- **Tests**
	- Enhanced test coverage for the `Note` component by integrating Storybook stories.
	- Added new test cases for `SingleNote`, `NoteWithCustomAuthor`, and `Editable` components.
	- Implemented accessibility checks using `axe` to ensure compliance with standards.
	- Maintained existing functionality and tests for the `Note` component.

### CODE REVIEW

File: src/components/Note/Note.test.tsx [hunk 1]
```diff
@@ -60,4 +67,89 @@ describe("Note", () => {
...
+			let renderResult: RenderResult;
+
+			beforeEach(() => {
+				renderResult = render(<SingleNote />);
+			});
+
+			it("should render ok", () => {
+				const { container } = renderResult;
+				expect(container).not.toBe(null);
+			});
```
Comment (joe-s-avaya) [inline src/components/Note/Note.test.tsx:82]: I don't think that these `expect(container).not.toBe(null);` do anything. We already have tests for basic rendering, and if the stories fail to render it'll blow up before this. Thus, I would recommend removing. Unless I'm misunderstanding something.

File: src/components/Note/Note.test.tsx [hunk 2]
```diff
@@ -60,4 +67,89 @@ describe("Note", () => {
...
+			it("toggles the textarea when edit is selected", async () => {
+				const { container } = renderResult;
+				await user.click(
+					container.querySelector("button") as HTMLButtonElement,
+				);
+				await user.click(container.getElementsByClassName("neo-icon-edit")[0]);
+
+				expect(
+					container.getElementsByTagName("textarea")[0],
+				).toBeInTheDocument();
```
Comment (joe-s-avaya) [inline src/components/Note/Note.test.tsx:131]: Kent C Dodds (creator of testing-library) recommends _not_ using `container`, and instead using `screen`. See [this blog post](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-using-screen) for details.

Comment (joe-s-avaya) [reply src/components/Note/Note.test.tsx:131]: We _must_ use `container` for the `const results = await axe(container); expect(results).toHaveNoViolations();` a11y checks (which I'm hoping to remove in favor of storybook's built in a11y testing in the future), but we should be using `screen.*` for all other queries unless absolutely necessary.

**Review by joe-s-avaya — APPROVED** [review]
I have some suggestions, and CodeRabbit has some suggestions, but nothing major. Approving, but please to review the suggestions.

**Review by yangAtSpoken — APPROVED** [review]

File: src/components/Note/Note.test.tsx [hunk 3]
```diff
@@ -60,4 +67,89 @@ describe("Note", () => {
...
+			});
```
[the remark this answers is not in the document]
Comment (razianmm) [reply src/components/Note/Note.test.tsx:132]: Thanks coderabbit, I've adopted your suggestion

File: src/components/Note/Note.test.tsx [hunk 4]
```diff
@@ -60,4 +67,89 @@
...
+				await user.click(
+					container.querySelector("button") as HTMLButtonElement,
+				);
+				await user.click(container.getElementsByClassName("neo-icon-edit")[0]);
+				await user.click(container.getElementsByTagName("textarea")[0]);
+				await user.keyboard("aa");
+				await user.click(container.getElementsByTagName("button")[1]);
+
+				expect(screen.getByText("Edited: Me")).toBeInTheDocument();
+			});
```
[the remark this answers is not in the document]
Comment (razianmm) [reply src/components/Note/Note.test.tsx:146]: Suggestion adopted

File: src/components/Note/Note.test.tsx (hunk 1 above)
Comment (razianmm) [reply src/components/Note/Note.test.tsx:82]: ✅

File: src/components/Note/Note.test.tsx (hunk 2 above)
Comment (razianmm) [reply src/components/Note/Note.test.tsx:131]: Removed along with adopting coderabbit suggestions ✅

**Pull request merged**